### PR TITLE
helm: support resources override for gateway injection

### PR DESCRIPTION
**Pull request by itsLucario**
Gateway injection currently doesn't support having configurable resource requests and limits. This PR uses similar annotations that are currently used by sidecar injection to override resources.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [X] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by itsLucario** [comment]
@jacob-delgado Is release note required for this? I see the check failing

**Comment by aleks-andr** [comment]
Would love to see this feature get merged. I had an almost identical [PR](https://github.com/istio/istio/pull/52694) open for this that's now closed due to being stale, hoping this one doesn't meet the same fate. 🙏

File: pkg/kube/inject/testdata/inject/gateway.yaml.injected [hunk 1]
```diff
@@ -111,7 +111,13 @@ spec:
             port: 15021
           periodSeconds: 15
           timeoutSeconds: 3
-        resources: {}
+        resources:
+          limits:
+            cpu: "2"
```
Comment (zirain) [inline pkg/kube/inject/testdata/inject/gateway.yaml.injected:116]: should we removed default cpu limit?
- https://medium.com/omio-engineering/cpu-limits-and-aggressive-throttling-in-kubernetes-c5b20bd8a718
- https://www.howtogeek.com/devops/should-you-set-kubernetes-cpu-limits/
- https://home.robusta.dev/blog/stop-using-cpu-limits

**Pull request closed without merging**